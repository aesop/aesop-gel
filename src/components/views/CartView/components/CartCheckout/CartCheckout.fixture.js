export default {
  copy: {
    checkoutButton: 'Checkout',
    checkoutButtonTitle: 'Go to Checkout',
    subTotal: 'Subtotal (Tax Incl.)',
  },
};
