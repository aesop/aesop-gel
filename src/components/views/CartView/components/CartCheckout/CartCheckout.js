import React from 'react';
import cx from 'classnames';
import Button from '~/components/elements/Button';
import Props from './CartCheckout.types';
import styles from './CartCheckout.module.css';

const CartCheckout: React.FunctionComponent<Props> = ({
  copy,
  className,
  subTotalPrice,
}) => {
  const classSet: string = cx(styles.base, className);

  return (
    <div className={classSet}>
      <div className={styles.subTotalContainer}>
        <div className={styles.subTotalLabel}>{copy.subTotal}</div>
        <div className={styles.subTotalPrice}>{subTotalPrice}</div>
      </div>
      <Button
        className={styles.checkoutButton}
        title={copy.checkoutButtonTitle}
        to="/checkout/"
      >
        {copy.checkoutButton}
      </Button>
    </div>
  );
};

export default CartCheckout;
