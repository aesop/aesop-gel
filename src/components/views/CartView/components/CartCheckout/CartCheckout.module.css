@import 'styles/variables/index';
@import 'styles/mixins/index';

.base {
}

.checkoutButton {
  width: 100%;
  padding-top: 21px;
  padding-bottom: 21px;
  background-color: var(--color-white-dark);
  color: var(--color-black);
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(15px);
  text-align: center;

  &:focus {
    outline-color: var(--color-grey-20);
  }

  &:focus,
  &:hover {
    background-color: var(--color-white);
    color: var(--color-black);
  }
}

.subTotalContainer {
  position: relative;
  padding: 30px 0 28px 0;
}

.subTotalLabel {
  color: var(--color-grey-60);
  font-family: var(--font-family-sans-serif-medium);
  font-size: rem(15px);
}

.subTotalPrice {
  position: absolute;
  top: 25px;
  right: 0;
  color: var(--color-white);
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(24px);
}
