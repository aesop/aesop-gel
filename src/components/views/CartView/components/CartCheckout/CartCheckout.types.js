import { ClassSetType } from '~/types';

interface CartCheckoutCopy {
  checkoutButton: string;
  checkoutButtonTitle: string;
  subTotal: string;
}

export interface Props extends ClassSetType {
  copy: CartCheckoutCopy;
  subTotalPrice: string;
}

export { Props as default }; // eslint-disable-line no-undef
