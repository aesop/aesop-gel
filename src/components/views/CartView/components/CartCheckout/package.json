{
  "name": "CartCheckout",
  "version": "0.0.0",
  "main": "CartCheckout.js"
}
