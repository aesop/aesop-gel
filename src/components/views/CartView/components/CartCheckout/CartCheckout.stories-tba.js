import React from 'react';
import { storiesOf } from '@storybook/react';
import 'normalize.css';
import '~/styles/base.module.css';
import '~/styles/storybook.module.css';
import CartViewFixture from '~/components/views/CartView/CartView.fixture';
import CartCheckoutFixture from './CartCheckout.fixture';
import CartCheckout from './CartCheckout';

storiesOf('Views.CartView.CartCheckout', module).add('Base component', () => (
  <CartCheckout
    copy={CartCheckoutFixture.copy}
    subTotalPrice={CartViewFixture.subTotalPrice}
  />
));
