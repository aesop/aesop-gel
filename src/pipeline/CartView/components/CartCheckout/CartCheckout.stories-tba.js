import React from 'react';
import { storiesOf } from '@storybook/react';
import 'normalize.css';

import '~/styles/base.module.css';
import '~.storybook/storybook.module.css';
import CartViewFixture from '~/components/CartView/CartView.fixture';
import CartCheckoutFixture from './CartCheckout.fixture';
import CartCheckout from './CartCheckout';

storiesOf('CartView.CartCheckout', module).add('Base component', () => (
  <CartCheckout
    copy={CartCheckoutFixture.copy}
    subTotalPrice={CartViewFixture.subTotalPrice}
  />
));
