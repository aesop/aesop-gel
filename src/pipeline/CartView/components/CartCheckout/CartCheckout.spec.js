import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import CartViewFixture from '~/components/CartView/CartView.fixture';
import CartCheckout from './CartCheckout';
import CartCheckoutFixture from './CartCheckout.fixture';

configure({ adapter: new Adapter() });

describe('<CartCheckout />', () => {
  it('should be defined', () => {
    expect(CartCheckout).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer
      .create(
        <CartCheckout
          copy={CartCheckoutFixture.copy}
          subTotalPrice={CartViewFixture.subTotalPrice}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
