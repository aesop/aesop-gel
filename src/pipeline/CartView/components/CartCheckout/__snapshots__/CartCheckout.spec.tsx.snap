// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCheckout /> renders base component correctly 1`] = `
<div
  className="base"
>
  <div
    className="subTotalContainer"
  >
    <div
      className="subTotalLabel"
    >
      Subtotal (Tax Incl.)
    </div>
    <div
      className="subTotalPrice"
    >
      $140.00
    </div>
  </div>
  <a
    className="base block internal checkoutButton"
    href="/checkout/"
    target="_self"
    title="Go to Checkout"
  >
    Checkout
  </a>
</div>
`;
